processing/src/main/java/org/apache/druid/segment/MultiValueConfigHandling.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment;
+
+import com.google.inject.Inject;
+import org.apache.druid.data.input.impl.DimensionSchema;
+
+/**
+ * Helper class to determine the default mode of multi value handling.
+ */
+public class MultiValueConfigHandling
+{
+  /**
+   * INSTANCE is injected using static injection.
+   */
+  @Inject
+  private static DefaultColumnFormatConfig INSTANCE;
```

Comment:
@clintropolis The static side effect + lifecycle register as done in `NestedDataModule` can work too, but I found this static injection to be slightly cleaner. What do you think?

Reply:
Yeah, I suppose I was thinking there's less boilerplate with static injection (but it turns out it's actually not that much). Also, the Guice [docs](https://github.com/google/guice/wiki/Injections#static-injections) suggests using static injections as a "crutch."

I have updated the config to now use the side effect pattern.